packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	comment as commentIcon,
+	commentAuthorAvatar as userIcon,
+	Icon,
+	check as resolvedIcon,
+} from '@wordpress/icons';
+import { useSelect } from '@wordpress/data';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+} from '@wordpress/components';
+import { dateI18n, format, getSettings } from '@wordpress/date';
+
+/**
+ * Internal dependencies
+ */
+import PluginSidebar from '../plugin-sidebar';
+import { store as editorStore } from '../../store';
+
+const isBlockCommentExperimentEnabled =
+	window?.__experimentalEnableBlockComment;
+
+/**
+ * Renders the Collab sidebar.
+ */
+export default function CollabSidebar() {
+	const { threads } = useSelect( ( select ) => {
+		const post = select( editorStore ).getCurrentPost();
+
+		return {
+			threads: post?.meta?.collab ? JSON.parse( post.meta.collab ) : [],
```

Comment:
We shouldn't return a new array instance on every select call. Otherwise this will excessively re-render when there is no meta and the store changes.

Reply:
Changed it